src/api/telemetry/TelemetryCollection.js

```diff
@@ -0,0 +1,358 @@
+            throw new Error(error);
+        }
+
+        this._processNewTelemetry(historicalData);
+
+    }
+    /**
+     * This uses the built in subscription function from Telemetry API
+     */
+    _initiateSubscriptionTelemetry() {
+
+        if (this.unsubscribe) {
+            this.unsubscribe();
+        }
+
+        this.unsubscribe = this.openmct.telemetry
+            .subscribe(this.domainObject, (datum) => {
+                this._processNewTelemetry(datum);
+            }, this.options);
+    }
+
+    /**
+     * call back for telemetry provider to add more data as well as
+     * pass in the current state of the telemetry collection
+     * (which the telemetry collection will hold)
+     *
+     *
+     * @param  {Object[]} telemetryData - array of telemetry data objects
+     * @param  {*} [collectionState] - providers can pass a collectionState that
+     * will be used for tracking between collection and provider
+     */
+    _addPage(telemetryData, collectionState) {
+        this.collectionState = collectionState;
+        this._processNewTelemetry(telemetryData);
+    }
+
+    /**
+     * Filter any new telemetry (add/page, historical, subscription) based on
+     * time bounds
+     *
+     * @param  {(Object|Object[])} telemetryData - telemetry data object or
+     * array of telemetry data objects
+     */
+    _processNewTelemetry(telemetryData) {
+        let data = Array.isArray(telemetryData) ? telemetryData : [telemetryData];
+        let parsedValue;
+        let beforeStartOfBounds;
+        let afterEndOfBounds;
+        let added = [];
+
+        for (let datum of data) {
+            parsedValue = this.parseTime(datum);
+            beforeStartOfBounds = parsedValue < this.lastBounds.start;
+            afterEndOfBounds = parsedValue > this.lastBounds.end;
+
+            if (!afterEndOfBounds && !beforeStartOfBounds) {
+                if (!this.boundedTelemetry.includes(datum)) {
+                    this.boundedTelemetry.push(datum);
+                    added.push(datum);
+                }
+            } else if (afterEndOfBounds) {
+                this.futureBuffer.push(datum);
+            }
+        }
+
+        if (added.length) {
+            this.emit('add', added);
+        }
+    }
+
+    /**
+     * when the start time, end time, or both have been updated.
+     * data could be added OR removed here we update the current
+     * bounded telemetry and emit the results
+     *
+     * @param  {TimeConductorBounds} bounds The newly updated bounds
+     * @param  {boolean} [tick] `true` if the bounds update was due to
+     * a "tick" event (ie. was an automatic update), false otherwise.
+     */
+    _bounds(bounds, isTick) {
+        let startChanged = this.lastBounds.start !== bounds.start;
+        let endChanged = this.lastBounds.end !== bounds.end;
+
+        this.lastBounds = bounds;
+
+        this.emit('bounds', ...arguments);
```

Comment:
Do we need this? We have a time API that tells us when bounds changes occur.

Reply:
Yeah I think this is in the same vein as "updateOptions" thought I saw a need for it, but we're not using it. I'll remove.